### PR TITLE
[Cherry-pick] Bump Thrift version from 0.13.0 to 0.14.1

### PR DESCRIPTION
**Pull request by RyoYang**
To bump the Thrift version to 0.14.1 [Cherry-pick from #9881]
- To avoid [CVE-2020-13949](https://nvd.nist.gov/vuln/detail/CVE-2020-13949)
- to fix some dependencies issues

- rename `src/thrfit_0_13_0` to `src/thrift_2` to remove version number in the path. (`src/thrift` contains rules to build thrift 0.11.0  )
- Add thrift sources as submodule as there are no prepared debian packages for version >0.13.0 on [debian.org](https://packages.debian.org/search?searchon=sourcenames&keywords=thrift)
- Added patches with fixes for original thrift debian rules:(remove unneeded packages, fix multi job build)
```
BLDENV=buster make -f Makefile.work target/debs/buster/libthrift-dev_0.14.1_amd64.deb
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Review by richardyu-ms — APPROVED** [review]

File: platform/broadcom/libsaithrift-dev.mk [hunk 1]
```diff
@@ -6,8 +6,8 @@ LIBSAITHRIFT_DEV = libsaithrift$(SAITHRIFT_VER)-dev_$(SAI_VER)_amd64.deb
 $(LIBSAITHRIFT_DEV)_SRC_PATH = $(SRC_PATH)/sonic-sairedis/SAI
 #Support two different versions of thrift
 ifeq ($(SAITHRIFT_V2),y)
-$(LIBSAITHRIFT_DEV)_DEPENDS += $(LIBTHRIFT_0_13_0) $(LIBTHRIFT_DEV_0_13_0) $(PYTHON3_THRIFT_0_13_0) $(THRIFT_COMPILER_0_13_0)
-$(LIBSAITHRIFT_DEV)_RDEPENDS += $(LIBTHRIFT_0_13_0)
+$(LIBSAITHRIFT_DEV)_DEPENDS += $(LIBTHRIFT_0_14_1) $(LIBTHRIFT_DEV_0_14_1) $(PYTHON3_THRIFT_0_14_1) $(THRIFT_COMPILER_0_14_1)
+$(LIBSAITHRIFT_DEV)_RDEPENDS += $(LIBTHRIFT_0_14_1)
```
[the remark this answers is not in the document]
Comment (RyoYang) [reply platform/broadcom/libsaithrift-dev.mk]: Yeah, we have noticed this typo in master branch, which will be fixed by introducing this PR. https://github.com/Azure/sonic-buildimage/pull/10199.

[the remark this answers is not in the document]
Comment (msosyak) [reply platform/broadcom/libsaithrift-dev.mk]: I see, but this PR is to 202111, so my point is that this PR should contain not only "[Cherry-pick] Bump Thrift version from 0.13.0 to 0.14.1" but also fix from https://github.com/Azure/sonic-buildimage/pull/10199

**Comment by msosyak** [comment]
@RyoYang Please resolve the conflicts

**Comment by RyoYang** [comment]
> @RyoYang Please resolve the conflicts

@msosyak This PR will be a draft and submit another PR (https://github.com/Azure/sonic-buildimage/pull/10528) to fix the conflict

**Pull request closed without merging**